core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -96,20 +97,38 @@ private[spark] class SparkSubmit extends Logging {
   }
 
   /**
-   * Kill an existing submission using the REST protocol. Standalone and Mesos cluster mode only.
+   * Kill an existing submission.
    */
   private def kill(args: SparkSubmitArguments): Unit = {
-    new RestSubmissionClient(args.master)
-      .killSubmission(args.submissionToKill)
+    if (RestSubmissionClient.supportsRestClient(args.master)) {
+      new RestSubmissionClient(args.master)
+        .killSubmission(args.submissionToKill)
+    } else {
+      val sparkConf = new SparkConf()
+      args.sparkProperties.foreach { case (k, v) => sparkConf.set(k, v) }
```

Comment:
I'd turn this into a new method `args.toSparkConf()`, since it's not copy & pasted in multiple places.